handbook/engineering/distribution/gitlab_native_local.md

```diff
@@ -0,0 +1,59 @@
+# How to test the Gitlab native integration on a local dev machine
+
+This specializes and augments the official admin docs for the integration with gotchas when running on localhost.
+It assumes you want to develop/debug the Sourcegraph side of this relationship so therefore will run the `dev/start.sh`
+Sourcegraph and will run a Gitlab CE instance in a docker container.
+
+## Sourcegraph instance
+
+Launch a Sourcegraph instance as usual with `dev/start.sh` or `enterprise/dev/start.sh`.
+
+If you need a TLS endpoint for the Sourcegraph instance (for example to debug secure-only cookies) you can use [Caddy](https://caddyserver.com/),
+ [ngrok](https://dashboard.ngrok.com/get-started) or similar to create the TLS endpoint and reverse-proxy the Sourcegraph instance.
+ Please don't forget to update the external URL of the Sourcegraph instance in the site config to the appropriate TLS endpoint value.
+ 
+## Gitlab instance
+
+It is assumed here that you have a functioning Docker runtime on your local dev machine. We will run Gitlab like so:
+
+```shell script
+docker run --hostname localhost -p 4080:4080  \
+        --name gitlab \
+        --volume /Users/exampledev/sourcegraph-docker/gitlab-disk/config:/etc/gitlab \
+        --volume /Users/exampledev/sourcegraph-docker/gitlab-disk/logs:/var/log/gitlab \
+        --volume /Users/exampledev/sourcegraph-docker/gitlab-disk/data:/var/opt/gitlab \
+        gitlab/gitlab-ce:latest
+```
+
+> Important: Make sure if you run Gitlab on a port different from port 80 then the container port has to match the host port
```

Comment:
If we're presuming someone is following your guide, then I don't think this is required.

Reply:
I just wanted to call this out in case they change ports. This bit me and took me awhile to figure out. If their host port in my command is already taken they should change both.